engine/minskyTensorOps.cc

```diff
@@ -393,6 +399,15 @@ namespace minsky
       hc.xvectors.insert(hc.xvectors.begin(), xv1.begin(), xv1.end());           
       cachedResult.hypercube(move(hc));
         
+      set<size_t> newIdx;
+      for (auto& i: arg1->index())
+        for (auto& j: arg2->index()) 
+        {
+            newIdx.insert(i+arg1->index().size()*j);
```

Comment:
This doesn't look right. The new index should be i+arg1->hypercube().numElements()*j, but obviously move the numElements() call outside the loop.

Reply:
Thanks, done!